core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2022,1459 @@ def export_data(
+        cls, language_code: str, contributor_id: str
+    ) -> Optional[TranslationReviewerTotalContributionStatsModel]:
+        """Gets the TranslationReviewerTotalContributionStatsModel
+        matching the supplied language_code, contributor_id.
+
+        Args:
+            language_code: str. ISO 639-1 language code.
+            contributor_id: str. User ID.
+
+        Returns:
+            TranslationReviewerTotalContributionStatsModel|None. The matching
+            TranslationReviewerTotalContributionStatsModel, or None
+            if no such model instance exists.
+        """
+        entity_id = cls.construct_id(
+            language_code, contributor_id)
+        return cls.get_by_id(entity_id)
+
+    @classmethod
+    def get_all_by_user_id(
+        cls, user_id: str
+    ) -> Sequence[TranslationReviewerTotalContributionStatsModel]:
+        """Gets all TranslationReviewerTotalContributionStatsModel
+        matching the supplied user_id.
+
+        Args:
+            user_id: str. User ID.
+
+        Returns:
+            list(TranslationReviewerTotalContributionStatsModel). The matching
+            TranslationReviewerTotalContributionStatsModel.
+        """
+        return cls.get_all().filter(
+            cls.contributor_id == user_id
+        ).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
+
+    @classmethod
+    def fetch_page(
+        cls,
+        page_size: int,
+        offset: int,
+        language_code: str,
+        sort_by: Optional[SortChoices],
+        num_days_since_last_activity: Optional[int]
+    ) -> Tuple[Sequence[TranslationReviewerTotalContributionStatsModel],
+                int,
+                bool]:
+        """Returns the models according to values specified.
+
+        Args:
+            page_size: int. Number of models to fetch.
+            offset: int. Number of results to skip from the beginning of all
+                results matching the query.
+            language_code: str. The language code to get results for.
+            sort_by: SortChoices|None. A string indicating how to sort the
+                result.
+            num_days_since_last_activity: int|None. To get number of users
+                who are active in num_days_since_last_activity.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models which match the supplied language_code,
+                    and num_days_since_last_activity filters, returned in the
+                    order specified by sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
```

Comment:
Same notes as before, for this and the comment below. Ditto for other classes.

Reply:
PTAL again. Thanks